### ISSUE TITLE
Sort by Multiple Columns

### ISSUE THREAD

**Issue by dmerand** [Airtable/airtable-ruby#7]
Hi,

Looking at the API docs, it seems that Airtable should be able to support sorting via multiple columns via the API:

> A **list of sort objects** that specifies how the records will be ordered. Each sort object must have a field key specifying the name of the field to sort on, and an optional direction key that is either "asc" or "desc". The default direction is "asc".

But I'm noticing that the `records` method (among others) in `lib/airtable/table.rb` seems to hard-code sort at one parameter:

```
    def records(options={})
      options["sortField"], options["sortDirection"] = options.delete(:sort) if options[:sort]
      results = self.class.get(worksheet_url, query: options).parsed_response
      RecordSet.new(results)
    end
```

I've tried simply commenting out the first line of that method and passing my own sort object (ie `sort: [{field: 'fieldName', direction: :asc}]`, but likely due to my unfamiliarity with HTTParty or the Airtable API, I can't figure out how to pass multiple sort objects through the `records` method.

Basically: I would have submitted a pull request to adjust this method to allow multiple sort, but I can't figure it out :/ Hopefully you know more than me on the topic.

Thanks!

**Comment by syrnick** [comment]
I haven't looked in the ruby code, but it seems that sort was an array with field and direction.

The sort should be json object like this:
```
[{"field": "Item", "direction": "desc"}]
```

**Comment by dmerand** [comment]
Thanks, yes that appears to be the format for sorting by a _single_ field. What I'm trying to figure out is how to sort by *multiple* fields.

For example, I want to sort a list of students by dormitory, then name, via the API. The best I've been able to come up with is to use a `view` in Airtable and set that view with a default multi-column sort. However that approach that doesn't work well when the data gets out-of-order (ie comes in randomly from a form) because I have to manually sort the form in Airtable before accessing it via my app. So I'd *love* to be able to sort multi-column directly via the API, I just can't figure out how to do it :/

**Comment by syrnick** [comment]
@dmerand can you try sortBy branch to see if that works for you?

**Comment by dmerand** [comment]
I tried the sortBy branch in my code, and it's working great! Thanks for taking the effort to make this update. It's very cool to be able to work directly on the returned object, rather than convert it to an array + sort it, which was my previous approach.

**Comment by Etsija** [comment]
Could the sortBy branch be merged and the gem be offered to everyone?  There's lots of us needing the multiple key sorting.

**Comment by Etsija** [comment]
...I think I was able to install the gem from GitHub from sortBy branch, but I cannot figure out how to use multiple keys.  Have tried so far:

@rec = @tbl.records(:sort => [["key1", :asc], ["key2", :asc]])

Could you give an example on how to use the multiple key sorting properly?

**Comment by syrnick** [comment]
@Etsija try sortByTake2 branch - it now handles :asc being a symbol.
